test/padding.jl

```diff
@@ -0,0 +1,78 @@
+@testset "padding constant" begin
+  x = rand(2, 2, 2)  
+  
+  y = @inferred pad_constant(x, (3, 2, 4, 5))
+  @test size(y) == (7, 11, 2)
+  @test y[4:5, 5:6, :] ≈ x
+  y[4:5, 5:6, :] .= 0
+  @test all(y .== 0)
+
+  @test pad_constant(x, (3, 2, 4, 5)) ≈ pad_zeros(x, (3, 2, 4, 5))
+  @test pad_zeros(x, 2) ≈ pad_zeros(x, (2,2)) 
+  
+  y = @inferred pad_constant(x, (3, 2, 4, 5), 1.2, dims=(1,3))
```

Comment:
I think `inferred` comes up as a problematic test often times, should be remove this?

Reply:
I don't know of such issues. I know of packages using inferred so much that their tests become very annoying to read https://github.com/JuliaGraphs/LightGraphs.jl/blob/ce533729381a6c0b0d8a1eb5f0c3e058833cb55b/test/core.jl
I think it is ok to have inferred when you worry the function could be type-unstable. However, in most code you are confident enough in type stability that it's totally ok to not check it with inferred. Since I was in doubt about this particular function, I preferer to keep the `@inferred` check.